edge/pkg/devicetwin/dttype/types_test.go

```diff
@@ -0,0 +1,623 @@
+/*
+Copyright 2019 The KubeEdge Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+   http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package dttype
+
+import (
+	"encoding/json"
+	"errors"
+	"reflect"
+	"testing"
+	"time"
+
+	"github.com/satori/go.uuid"
+)
+
+// Test_SetEventID is function to test SetEventID().
+func Test_SetEventID(t *testing.T) {
+	tests := []struct {
+		name      string
+		timestamp int64
+		eventID   string
+	}{
+		{
+			name:    "SetEventIDTest",
+			eventID: uuid.NewV4().String(),
+		},
+	}
+	for _, test := range tests {
+		t.Run(test.name, func(t *testing.T) {
+			bs := BaseMessage{}
+			bs.SetEventID(test.eventID)
+			if bs.EventID != test.eventID {
+				t.Errorf("Got wrong EventID, Got = %v, Want = %v", bs.EventID, test.eventID)
+			}
+		})
+	}
+}
+
+// TestBuildBaseMessage is function to test BuildBaseMessage().
+func TestBuildBaseMessage(t *testing.T) {
+	tests := []struct {
+		name string
+	}{
+		{
+			name: "BuildBaseMessageTest",
+		},
+	}
+	for _, test := range tests {
+		t.Run(test.name, func(t *testing.T) {
+			got := BuildBaseMessage()
+			if got.EventID == "" {
+				t.Errorf("BuildBaseMessage() failed,Failed to generate EventID")
+			}
+			if got.Timestamp == 0 {
+				t.Errorf("BuildBaseMessage() failed,Failed to get timestamp")
+			}
+		})
+	}
+}
+
+// devTwin() is function to create a MsgTwin and add it to map with given deviceID.
+func devTwin(deviceID string) map[string]*MsgTwin {
```

Comment:
it does add the type as deleted . the function name and comment should reflect / convey the same.

Reply:
Updated